dashboard/src/components/Config/AppRepoList/AppRepoForm.tsx

```diff
@@ -0,0 +1,259 @@
+    customCA: "",
+  };
+
+  public render() {
+    return (
+      <form className="container padding-b-bigger" onSubmit={this.handleInstallClick}>
+        <div className="row">
+          <div className="col-12">
+            <div>
+              <h2>Add an App Repository</h2>
+            </div>
+            <div>
+              <label>
+                <span>Name:</span>
+                <input
+                  type="text"
+                  placeholder="example"
+                  value={this.state.name}
+                  onChange={this.handleNameChange}
+                  required={true}
+                  pattern="[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*"
+                  title="Use lower case alphanumeric characters, '-' or '.'"
+                />
+              </label>
+            </div>
+            <div>
+              <label>
+                <span>URL:</span>
+                <input
+                  type="url"
+                  placeholder="https://charts.example.com/stable"
+                  value={this.state.url}
+                  onChange={this.handleURLChange}
+                  required={true}
+                />
+              </label>
+            </div>
+            <div>
+              <label>
+                <span>Authorization (optional):</span>
+                <div className="row">
+                  <div className="col-3">
+                    <label className="margin-l-big">
+                      <input
+                        type="radio"
+                        name="auth"
+                        value="None"
+                        defaultChecked={true}
+                        onChange={this.handleAuthRadioButtonChange}
+                      />
+                      None
+                      <br />
+                    </label>
+                    <label>
+                      <input
+                        type="radio"
+                        name="auth"
+                        value="Basic"
+                        onChange={this.handleAuthRadioButtonChange}
+                      />
+                      Basic Auth
+                      <br />
+                    </label>
+                    <label>
+                      <input
+                        type="radio"
+                        name="auth"
+                        value="Bearer"
+                        onChange={this.handleAuthRadioButtonChange}
+                      />
+                      Bearer
+                      <br />
+                    </label>
+                    <label>
+                      <input
+                        type="radio"
+                        name="auth"
+                        value="HTTP Header"
+                        onChange={this.handleAuthRadioButtonChange}
+                      />
+                      Custom
+                      <br />
+                    </label>
+                  </div>
+                  <div className="col-9">
```

Comment:
Also, since these elements are going to appear and disappear, please add the `aria-live="polite` property to notify screen readers about these changes.